### PR TITLE
Adjustments to lib/kim/Install.py and docs

### LINKED ISSUE #3042: more flexible behavior from KIM package compilation script
Our cmake executable is named `cmake3`, but this https://github.com/lammps/lammps/blob/1a4511bb8db2ec6555e82805f7e015b08bd4248d/lib/kim/Install.py#L144 hardwires just `cmake`.  I made a symlink, but it might be nice for that to be user controllable, e.g. via a `CMAKE` env or make var.

Also, it would be useful to make cmake use the same compilers as the rest of the LAMMPS compilation if it compiles a .  Ours, when run by the `Install.py` script,  seems to default to incorrect ones (seemed to be something conda related, even though we used `args="-b"` to compile a local copy of kim).  I'll try to give a reproducible process to show this behavior and post here.

### PR DESCRIPTION
**Pull request by ellio167**
**Summary**

Add `CMAKE` env var support and document

**Related Issue(s)**

closes #3042 

**Author(s)**

Ryan S Elliott (UMN)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

n/a

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

File: doc/src/Build_extras.rst [hunk 1]
```diff
@@ -341,6 +341,18 @@ minutes to hours) to build.  Of course you only need to do that once.)
...
+      cmake build system.  The ``lib/kim/Install.py`` script supports a
+      ``CMAKE`` environment variable if the cmake executable is named other
+      than ``cmake`` on your system.  Additional environment variables may be
+      provided for use by camke on the command line.  For example, to use the
+      ``cmake3`` executable and tell it to use the gnu version 11 compilers
+      to build KIM, one could use the following command line.
+
+      .. code-block:: bash
+
+         $ CMAKE=cmake3 CXX=g++-11 CC=gcc-11 FC=gfortran-11 make lib-kim args="-b "  # (re-)install KIM API lib using cmake3 and gnu v11 compilers with only example models
```
Comment (bernstei) [inline doc/src/Build_extras.rst:354]: Is this the preferred way to specify compilers, or is it something like `CMAKE_FC_COMPILER`?

Comment (ellio167) [reply doc/src/Build_extras.rst:354]: It is "Method 1" from the cmake FAQ

https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-do-i-use-a-different-compiler

**Comment by bernstei** [comment]
OK - I've seen conflicting information as to which set of env vars is preferred.

File: doc/src/Build_extras.rst [hunk 2]
```diff
@@ -341,6 +341,18 @@ minutes to hours) to build.  Of course you only need to do that once.)
          $ make lib-kim args="-p /usr/local" # use an existing KIM API installation at the provided location
          $ make lib-kim args="-p /usr/local -a EAM_Dynamo_Ackland_W__MO_141627196590_002" # ditto but add one model or driver
 
+      When using the "-b " option, the KIM library is built using its native
+      cmake build system.  The ``lib/kim/Install.py`` script supports a
+      ``CMAKE`` environment variable if the cmake executable is named other
+      than ``cmake`` on your system.  Additional environment variables may be
+      provided for use by camke on the command line.  For example, to use the
```
Comment (yafshar) [inline doc/src/Build_extras.rst:348]: @ellio167  minor typo in `provided for use by camke` and it should be corrected to `provided for use by the cmake`

**Review by sjplimp — APPROVED** [review]

**Pull request merged**